plugins/lua/sort/places.lua

```diff
@@ -0,0 +1,121 @@
+    if zone_type == df.civzone_type.WaterSource then return 'Water source' end
+    if zone_type == df.civzone_type.SandCollection then return 'Sand' end
+    if zone_type == df.civzone_type.FishingArea then return 'Fishing' end
+    if zone_type == df.civzone_type.Pond then return 'Pit/pond' end
+    if zone_type == df.civzone_type.Pen then return 'Pen/pasture' end
+    if zone_type == df.civzone_type.ClayCollection then return 'Clay' end
+    if zone_type == df.civzone_type.AnimalTraining then return 'Animal training' end
+    if zone_type == df.civzone_type.PlantGathering then return 'Gather fruit' end
+    if zone_type == df.civzone_type.Dungeon then return 'Dungeon' end
+    if zone_type == df.civzone_type.MeetingHall then return 'Meeting area' end
+    if zone_type == df.civzone_type.Barracks then return 'Barracks' end
+    if zone_type == df.civzone_type.ArcheryRange then return 'Archery range' end
+    if zone_type == df.civzone_type.Office then return 'Office' end
+    if zone_type == df.civzone_type.DiningHall then return 'Dining hall' end
+    if zone_type == df.civzone_type.Dormitory then return 'Dormitory' end
+    if zone_type == df.civzone_type.Bedroom then return 'Bedroom' end
+    if zone_type == df.civzone_type.Tomb then return 'Tomb' end
+
+    return '' -- If zone_type is anything else it will not have a default name populated in the Zones page when left nameless
+end
+
+local function get_zone_search_key(zone)
+    local site = df.global.world.world_data.active_site[0]
+    local result = {}
+
+    -- allow zones to be searchable by their name
+    if zone.name == nil or zone.name == '' then
+        table.insert(result, get_default_zone_name(zone.type))
+    else
+        table.insert(result, zone.name)
+    end
+
+    -- allow zones w/ assignments to be searchable by their assigned unit
+    if zone.assigned_unit ~= nil then
+        table.insert(result, dfhack.TranslateName(zone.assigned_unit.name))
+        table.insert(result, dfhack.units.getReadableName(zone.assigned_unit))
+    end
+
+    -- allow zones to be searchable by type
+    if zone.location_id == -1 then -- zone is NOT a special location and we don't need to do anything special for type searching
+        table.insert(result, df.civzone_type[zone.type]);
+    else -- zone is a special location and we need to get its type from world data
+        table.insert(result, dfhack.TranslateName(site.buildings[zone.location_id].name, true))
+        table.insert(result, df.civzone_type[site.buildings[zone.location_id].name.type])
+    end
+
+    -- allow barracks to be searchable by assigned squad
+    for _, squad in ipairs(zone.squad_room_info) do
+        table.insert(result, dfhack.military.getSquadName(squad.squad_id))
+    end
+
+    return table.concat(result, ' ')
+end
+
+-- ----------------------
+-- PlacesOverlay
+--
+
+PlacesOverlay = defclass(PlacesOverlay, sortoverlay.SortOverlay)
+PlacesOverlay.ATTRS{
+    default_pos={x=71, y=8},

```

Comment:
I feel like y=8 is too high but y=9 is too low.
I went with y=8 because it at least lined up with the edges of some of the other UI elements.

Reply:
finding a spot can be tricky. could you post a screenshot of what it looks like now?

Also, be sure the position works (or is adjusted) when the screen is resized narrower and the top row of tabs becomes two lines

Reply:
Here is y=8:
![image](https://github.com/DFHack/dfhack/assets/46685109/f3f4b3ed-ccd1-4895-86aa-b7b8c1466d72)

And yeah it looks fine when compressed:
![image](https://github.com/DFHack/dfhack/assets/46685109/de86f0b1-dc4d-4910-8724-061b55bc2fca)

Here is y=9 for reference:
![image](https://github.com/DFHack/dfhack/assets/46685109/f601b5e6-7117-4660-95e3-7c1e91516420)

Now that I'm looking at them side by side maybe y=9 is actually better since it's less likely to have misclicks onto the top bar? As much as it's bothersome that it doesn't line up.

Reply:
agreed. I think I chose a y=9 style for other searches as well